yarn-project/circuit-types/src/tx/processed_tx.ts

```diff
@@ -297,7 +297,11 @@ function validateProcessedTxLogs(tx: ProcessedTx): void {
     );
   }
   const unencryptedLogs = tx.unencryptedLogs || UnencryptedTxL2Logs.empty();
-  kernelHash = tx.data.end.unencryptedLogsHash;
+  kernelHash = Fr.fromBuffer(
+    UnencryptedTxL2Logs.hashSiloedLogs(
+      tx.data.end.unencryptedLogsHashes.filter(hash => !hash.isEmpty()).map(h => h.getSiloedHash()),
+    ),
+  );
```

Comment:
I find it weird that this `validateProcessedTxLogs` is called in the public processor, after it has processed a tx with or without public calls.

Shouldn't the check for log hashes vs log data be done in the tx validator? A tx shouldn't be considered valid if the user submits wrong log data for it.

The public processor doesn't have to check the logs after simulation if the check is already done before the tx is given to it. If there's any unencrypted log emitted from the public functions, the data will be there. Unless it's using a faulty simulator.

Maybe it's a question for @just-mitch? Wonder if it makes sense or am I missing something 😅

Reply:
Yes, I think it would need to be to be a tx validator since with the current approach the sequencer will select the TX for inclusion even if the preimages of the logs don't match the hashes in the kernels!